### ISSUE TITLE
Can't remove all subscriptions or backup only memorized video

### ISSUE THREAD

**Issue by thiswillbeyourgithub** [ram-on/SkyTube#197]
Hi,
Today I unsubscribed from a bunch of channels directly on my youtube account with the official website and I intended to export the xml file and then re import it to skytube in order to have the correct list of channels on the app.

The problem is : I can't seem to remove what I previously imported list of channels to then import the brand new list. All I can do is removing the app's data, and doing so removes the list of memorized video. Also the option to backup the memorized videos unfortunately also backs up the list of followed channels.

I am not sure my problem is clear but the simplest fix would be to turn the "backup channels and memorized videos" to two button "backup channels" and "backup memorized videos". Same for import.

**Comment by ram-on** [comment]
So basically you want an option to unsubscribe from all channels, right?

**Comment by atomjack** [comment]
Seems like importing from the youtube xml file should give the option of clearing out the subscriptions channels.

**Comment by thiswillbeyourgithub** [comment]
@atomjack that would do it. Although I think it might be very usefull to be able to backup the "saved videos" and the list of channel subscriptions to two separate files.

**Comment by okan35** [comment]
@atomjack  @ram-on Has anybody proceeded on this ? Otherwise I can start to work on this.

**Comment by ram-on** [comment]
@okan35 Go for it.  Leave the option to backup both  subscriptions and bookmarks DBs.

**Comment by okan35** [comment]
@ram-on So I put a checkbox like this ibb.co/i4ZEnc  When a user clicks it and proceeds to import, the user will be unsubscribed from all the channels. If this is what is want I will finish this and make a pull request soon.

**Comment by ram-on** [comment]
@okan35 Looks great - pls proceed

**Comment by okan35** [comment]
@ram-on I am able to unsubscribe a user from all the channels but I am not able to update SubsAdapter.  I made an Asynctask as I get `NetworkOnMainThread` error. So I made this inside `SubscriptionsDb` class. 

```java
 class deleteAllRecordsTask extends AsyncTask<YouTubeChannel,Void,Void>{

	  @Override
	 protected Void doInBackground(YouTubeChannel... youTubeChannels) {

		 try {
			 Log.d("UNNN", "unsubscribeFromAllChannels: " + SubscriptionsDb.getSubscriptionsDb().getTotalSubscribedChannels());
			 SubsAdapter subsAdapter; // here I am not sure how to instante
List<YouTubeChannel>channelList=SubscriptionsDb.getSubscriptionsDb().getSubscribedChannels();
			 for (YouTubeChannel youTubeChannel : channelList){
					SubscriptionsDb.getSubscriptionsDb().unsubscribe(youTubeChannel);
					subsAdapter.removeChannel(youTubeChannel);

			 }
			 Log.d("UNNN", "unsubscribeFromAllChannels: " + SubscriptionsDb.getSubscriptionsDb().getTotalSubscribedChannels());
		 } catch (IOException e) {
			 e.printStackTrace();
		 }
		  return null;
	 }
 }
```

My problem is I am not sure how to initialize SubsAdapter in the code. I tried using context, putting it in constructor of `AsyncTask` and then in `SubscriptionsDb# unsubscribeFromAllChannels()` using context of SubscriptionsDb.

```java
public void unsubscribeFromAllChannels()  {
		new deleteAllRecordsTask().execute();
	}```
but I get another error `  Caused by: android.view.ViewRootImpl$CalledFromWrongThreadException: Only the original thread that created a view hierarchy can touch its views.`

So I would like to hear if you have any suggestions how to use SubsAdapter in the AsyncTask so after unsubscribing from all channels I can use the `SubsAdapter#removeChannel`.

**Comment by ram-on** [comment]
Hello @okan35, just call `SubsAdapter#removeChannel()` on `AsynchTaskParallel#onPostExecute()`.  Remember, UI code can only be modified by the main thread, that is if your using an AsynchTask, UI code can be placed inside `onPreExecute()`, `onPostExecute()` and `onProgressUpdate()`.

On a side note, pls give a meaning full name to classes.  `deleteAllRecordsTask` name is meaningless to me.

Also, exceptions should be logged as follows:  `Logger.e(this, "An error occurred while XXXXXX", exception);`

**Comment by okan35** [comment]
@ram-on I fixed it by a `Handler` to pass to UI thread. The moment user checks all the subscriptions are unsubscribed. I will also fix the logs and name as they were just for the moment.

**Comment by ram-on** [comment]
@thiswillbeyourgithub Implemented as per PR #264
